src/components/CalendarGroups/CalendarGroups.js

```diff
@@ -82,12 +84,30 @@ const CalendarGroups = ({
               date={selectedDate}
               key={`groupColumn${index}`}
               columnKey={`groupColumn${index}`}
-              columnIndex={index}
+              columnIndex={columnIndex}
               columnId={eventGroup.id}
               eventsForColumn={eventsForDay}
               stepDetailsForColumn={stepDetailsForDay}
               getUpdatedDraggedEvent={({ event, columnMoves }) => {
-                const newIndex = index + columnMoves;
+                let newIndex = index + columnMoves;
+
+                if (visibleEventGroups) {
+                  let count = 1;
+                  if (columnMoves < 0) {
+                    count = -1;
+                  }
+                  while (
+                    !visibleEventGroups.includes(eventGroups[newIndex].id)
+                  ) {
+                    newIndex = index + columnMoves + count;
+                    if (columnMoves < 0) {
+                      count -= 1;
+                    } else {
+                      count += 1;
```

Comment:
I'm not sure what each of these variables actually represent - maybe a comment could be added to explain what's happening here?